server/src/main/java/org/opensearch/identity/StringPrincipal.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.identity;
+
+import java.security.Principal;
+import java.util.Objects;
+
+/**
+ * Create a principal from a string
+ *
+ * @opensearch.experimental
+ */
+public class StringPrincipal implements Principal {
```

Comment:
Should we rename it to `UsernamePrincipal` instead? `StringPrincipal` is definitely not giving the context what this principal is about.

Reply:
Ok, what about `NamedPrincipal`?